jsbits/Scenario.js

```diff
@@ -112,24 +112,28 @@ function gm$updateProps(bArray, values) {
  * Returns nicely sorted points, lines, surfaces + deletes and errors.
  * Add geomID to all features that miss it.
  *
- * @param fc - FeatureCollection
- * @returns {[points:Feature,lines:Feature,surfaces:Feature,deletes:Number(geomID),errors:string,cmin,cmax,dims]}
+ * @param gi - Geometry Input JSON, with FeatureCollection as one of its children objects.
+ * @returns {[points:Feature,lines:Feature,surfaces:Feature,deletes:Number(geomID),errors:string,cmin,cmax,dims,lat,lon,alt,srid]}
  */
-function gm$smartProcessFeatureCollection(fc, defVec, maxGeomId) {
+function gm$smartProcessFeatureCollection(gi, defVec, maxGeomId) {
```

Comment:
As I see, you changed it to read a JSON file with a FeatureCollection in geometry field.
I would propose to keep this function parsing a bare FeatureCollection instead.
Also, add an optional parameter to decide whether this function needs to guess WGS'84/metric or it is known from the root JSON